### PR TITLE
8317452: [JVMCI] Export symbols used by lightweight locking to JVMCI compilers.

### PR DESCRIPTION
**Pull request by mur47x111**
Export JavaThread::_lock_stack, LockStack::_top, LockStack::_end_offset, and ObjectMonitor::ANONYMOUS_OWNER to JVMCI compilers. Delete JVMCIUseFastLocking option

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8317452](https://bugs.openjdk.org/browse/JDK-8317452): [JVMCI] Export symbols used by lightweight locking to JVMCI compilers. (**Enhancement** - P4)

### Reviewers
 * [Doug Simon](https://openjdk.org/census#dnsimon) (@dougxc - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16032/head:pull/16032` \
`$ git checkout pull/16032`

Update a local copy of the PR: \
`$ git checkout pull/16032` \
`$ git pull https://git.openjdk.org/jdk.git pull/16032/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16032`

View PR using the GUI difftool: \
`$ git pr show -t 16032`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16032.diff">https://git.openjdk.org/jdk/pull/16032.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16032#issuecomment-1745671666)

### CODE REVIEW

File: src/hotspot/share/jvmci/jvmci_globals.hpp [hunk 1]
```diff
@@ -127,9 +127,6 @@ class fileStream;
   product(bool, JVMCICountersExcludeCompiler, true, EXPERIMENTAL,           \
           "Exclude JVMCI compiler threads from benchmark counters")         \
                                                                             \
-  develop(bool, JVMCIUseFastLocking, true,                                  \
```
Comment (dougxc) [inline src/hotspot/share/jvmci/jvmci_globals.hpp:130]: You should just delete this option - JVMCI is clearly marked as experimental.
What you have in this PR is not proper deprecation. What if someone specifies `-XX:-JVMCIUseFastLocking`? They will get no warning or VM exit.

Comment (mur47x111) [reply src/hotspot/share/jvmci/jvmci_globals.hpp:130]: Indeed. I have deleted the option.

**Review by dougxc — APPROVED** [review]

**Review by dougxc — APPROVED** [review]

**Pull request closed without merging**